### DISCUSSION TITLE
Why ProblemDetails class has additional 's' in its name?

### DISCUSSION THREAD

**Discussion by NiceStepUp** [dotnet/aspnetcore#48646] [General] [answered]
We have a debate about whether it is appropriate to have `s` or plural names in a class name. So as a guidance we took a look at [ProblemDetails](https://github.com/dotnet/aspnetcore/blob/main/src/Http/Http.Abstractions/src/ProblemDetails/ProblemDetails.cs) class which has `s`.

According to this [msdn docs](https://learn.microsoft.com/en-us/previous-versions/windows/desktop/automat/use-the-correct-plural-for-the-class-name) and if I understood this doc correctly, then name should be `ProblemDetail`, not `ProblemDetails`.

Can anybody explain to me why `s` is introduced there? I just want to keep to the best practice

**Comment by gfoidl** [comment, marked as answer]
> is appropriate to have `s` or plural names in a class name

It's appropriate if the class represents something that is in plural. So why not?
For example look at https://grep.app/search?q=class%20%5Cw%2Bs%20&regexp=true&filter[lang][0]=C%23 which shows quite some class names ending with `s`.
That search also shows a basic .NET type -- especially in the event based world -- `EventArgs` with a plural name (as it can have more than one actual args).

I think `ProblemDetails` is correct, as it shows the details of the problem, which can be more than one -- so plural.

But note: naming is one of the hardest problem in software 😉
So choose a name that suits best your needs, and don't stick too hard to generic rules. When you're working in a team, maybe within the team you can determine your own rules that are followed along.
